server/src/main/java/org/elasticsearch/transport/OutboundHandler.java

```diff
@@ -99,14 +90,9 @@ void sendRequest(final DiscoveryNode node, final TcpChannel channel, final long
      * @see #sendErrorResponse(Version, TcpChannel, long, String, Exception) for sending error responses
      */
     void sendResponse(final Version nodeVersion, final TcpChannel channel, final long requestId, final String action,
-                      final TransportResponse response, final boolean compressResponse, final boolean isHandshake) throws IOException {
+                      final TransportResponse response, final Compression.Scheme compressionScheme, final boolean isHandshake)
+        throws IOException {
         Version version = Version.min(this.version, nodeVersion);
-        final Compression.Scheme compressionScheme;
```

Comment:
I wonder if we can now assert that if `compressionScheme == LZ4` we are on a new enough version. Something like:

```
assert compressionScheme != Compression.Scheme.LZ4 || version.onOrAfter(Compression.Scheme.LZ4_VERSION);
```

I am sure this will break a test or two and I am OK with deferring this if it adds too much trouble.